### DISCUSSION TITLE
SH-RPi is compatible with Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by mairas** [hatlabs/discussions#79] [General]
Hi all,

I finally got my hands on a Raspberry Pi 5. I gave it a quick spin together with the Sailor Hat for Raspberry Pi v2, and I'm happy to report that the SH-RPi works fine with it! Even the Raspberry Pi 5 Active Cooler (which I recommend) fits with no modifications.

At the moment, there are a couple of minor software issues. First, it seems the installer script fails to enable I2C automatically, so you need to first enable it using `sudo raspi-config` before running the installer. And second, since the Pi does not detect a valid USB power supply connected to the USB-C jack, it limits the maximum peripheral current to 0.6 A. This can be circumvented by inserting the following line to `/boot/firmware/config.txt`

    usb_max_current_enable=1

**Comment by mairas** [comment]
I did a quickie stress test to evaluate the feasibility of powering a Pi 5 over NMEA 2000. The Pi 5 is a hungry beast. Together with an SH-RPi at 100% CPU load, they already pull 0.78A @ 11.2V (For this test, I'm using the first crummy 12V power supply I got my hands on.) This is very close to the roughly 0.8A default current limit setting on the SH-RPi, and if the supply voltage drops any further, the input current limiter would kick in and start depleting the supercaps. And this is without any peripherals whatsoever. So, in essence, if you want to use a Pi 5 and a SH-RPi on a boat, you should provide a dedicated power input for them.

**Reply by RichFind** [reply]
Not sure what you mean by "So, in essence, if you want to use a Pi 5 and a SH-RPi on a boat, you should provide a dedicated power input for them."

What does them refer to?

Are you saying that you cannot run the RPi from the SH-RPi connected directly to the boat's battery?  

The reason being if the SH-RPi is connected to the boat's 12 V battery and the battery voltage is too low the input current limiter would kick in and deplete the caps?

Is there enough power available to run an external SSD on the RPi USB port?

**Reply by mairas** [reply]
That sentence could've been phrased better... What I wanted to say is that with a Raspberry Pi 5, you shouldn't power the device using an NMEA 2000 network but instead with dedicated power cable. Powering from the boat battery is of course OK.

A low power condition is detected if the input voltage drops below 10V after the polarity protection diode, meaning the battery voltage is below about 10.6 V. This level is configurable. That condition always results in an initiated shutdown to avoid depleting the batteries any further.

An external USB SSD is generally not a problem. For boat use, you should pick a low-power one, however. The latest and greatest desktop and server SSDs with huge heatsinks might indeed require a lot of current, and since a Pi can't utilize their bandwidth, it'd be a wasted effort.

**Comment by RichFind** [comment]
Change in location of config.exe

When I enter  sudo nano /boot/config.txt
I get the message

DO NOT EDIT THIS FILE

The file you are looking for has moved to /boot/firmware

sudo nano /boot/firmware/config.txt

**Reply by mairas** [reply]
Indeed! Fixed. Thanks for the correction.

**Comment by RichFind** [comment]
The RPi 5 has a real time clock. Should I use the RTC on the Pi or the SH-RPi?
Based on the question immediately above, do you have any guidance about how to disable the unneeded RTC?

**Reply by mairas** [reply]
The SH-RPi clock can be programmed to restart the device e.g. for periodic checks and data acquisition and it also has a built-in battery holder.

On the other hand, the RTC on Pi 5 might be more carefully calibrated, resulting in lower clock drift. That's usually not a major issue, though, since the drift will be corrected when the device becomes connected to the internet or when GPS time becomes available (assuming Signal K Server is used).

I'd go with whichever is the easiest for you.

If you have previously enabled the SH-RPi RTC, you can disable it by commenting out the following line in `/boot/firmware/config.txt`: 

    dtoverlay=i2c-rtc,pcf8563

Alternatively, if you want to disable the Pi 5 RTC, add the following line:

    dtparam=rtc=off

**Reply by RichFind** [reply]
Where is the link to the installer?

**Reply by mairas** [reply]
https://docs.hatlabs.fi/sh-rpi/docs/getting-started/#software-installation
